### DISCUSSION TITLE
validation error for class-maps - vlan variable (integer instead of string?)

### DISCUSSION THREAD

**Discussion by rradim** [aristanetworks/avd#3165] [Q&A]
Hello,

it seems to me that class-map vlan variable is not defined correctly. Based on documentation it requires INTEGER. But since EOS supports definition of vlan range. It should probably be a string...

https://avd.sh/en/stable/roles/eos_cli_config_gen/docs/tables/class-maps.html?h=class_map
<html><body>

vlan | Integer |   |   |   | VLAN value(s) or range(s) of VLAN values
-- | -- | -- | -- | -- | --

</body>
</html>

My example is:

```
class_maps:
  qos:
    - name: cmap-access-vlans-653-654
      vlan: 653-654
```

It all works with the range, but it brings a validation error:

`[WARNING]: [XXXXX]: 'Validation Error: class_maps.qos[2].vlan': '653-654' is not of type 'int'`

Is there another way how to define the class map to avoid this validation error? For now I ignore it..but its not nice.

Thanks
Radim

**Comment by ClausHolbechArista** [comment]
Thank you for reporting this. We will fix it.

**Comment by ClausHolbechArista** [comment]
Fixed in #3215 which is included in the new 4.4.0 version we just released. Closing here, but feel free to reopen if something is still wrong.
